xilem_core/src/views/fork.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use core::marker::PhantomData;
+
+use crate::{
+    AppendVec, ElementSplice, Mut, NoElement, View, ViewId, ViewPathTracker, ViewSequence,
+};
+
+/// Create a view which acts as `active_view`, whilst also running `alongside_view`, without inserting it into the tree.
+///
+/// `alongside_view` must be a `ViewSequence` with an element type of [`NoElement`].
+pub fn fork<Active, Alongside, Marker>(
+    active_view: Active,
+    alongside_view: Alongside,
+) -> Fork<Active, Alongside, Marker> {
```

Comment:
I wonder whether this should have all the bounds necessary for `View` to avoid misusage?

Reply:
This has problems with ambiguity. Essentially, there is no link between the method call and the implementation of `ViewSequence` which gets selected.

I think it is possible to work around this, but it requires adding `State` and `Action` generics to each view which uses this trick. The same thing applies to e.g. `label`, for example.

I think we could do this, but it should happen in a follow-up and be applied to all views

Reply:
Sounds good to me, and reflects what I have commented in #428. (Though that issue there might not occur here, I'm not sure though).